EventFilter/HcalRawToDigi/src/HcalDataFrameFilter.cc

```diff
@@ -92,6 +129,29 @@ ZDCDigiCollection HcalDataFrameFilter::filter(const ZDCDigiCollection& incol, Hc
   return output;
 }
 
+QIE10DigiCollection HcalDataFrameFilter::filter(const QIE10DigiCollection& incol, HcalUnpackerReport& r) {
+  QIE10DigiCollection output(incol.samples());
+  for (QIE10DigiCollection::const_iterator i=incol.begin(); i!=incol.end(); i++) {
+    if (!HcalDataFrameFilter_impl::checkQIE10(*i,requireCapid_,requireDVER_))
+      r.countBadQualityDigi(i->id());
+    // Never exclude QIE10 digis as their absence would be
+    // treated as a digi with zero charged deposited in that channel
+    output.push_back(*i);
```

Comment:
please remind me where the quality checks are done.
Saving everything means that the filter is not functional and that bad channels or frames will show up downstream. [perhaps the reason that we had recently added PRs which ignore invalid hits is a reflection of this hack].

Also, why is bad digi with possibly garbage content better than an implied zero digi?